src/features/visiting/house/House.tsx

```diff
@@ -0,0 +1,222 @@
+import React, { useContext } from "react";
+import { useActor } from "@xstate/react";
+import { Modal } from "react-bootstrap";
+
+import { Context } from "features/game/VisitingProvider";
+import { GRID_WIDTH_PX } from "features/game/lib/constants";
+import { getLevel, getRequiredXpToLevelUp } from "features/game/types/skills";
+
+import house from "assets/buildings/house.png";
+import smoke from "assets/buildings/smoke.gif";
+import player from "assets/icons/player.png";
+import questionMark from "assets/icons/expression_confused.png";
+import close from "assets/icons/close.png";
+
+import plant from "assets/icons/plant.png";
+import pickaxe from "assets/tools/stone_pickaxe.png";
+
+import { Action } from "components/ui/Action";
+import { InnerPanel, OuterPanel, Panel } from "components/ui/Panel";
+import { Label } from "components/ui/Label";
+import { ITEM_DETAILS } from "features/game/types/images";
+import { InventoryItemName } from "features/game/types/game";
+
+import { SkillTree } from "./components/SkillTree";
+import { homeDoorAudio } from "lib/utils/sfx";
+import { Button } from "components/ui/Button";
+
+export const House: React.FC = () => {
+  const { gameService } = useContext(Context);
+  const [gameState] = useActor(gameService);
+  const { state } = gameState.context;
+
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [isSkillTreeOpen, setIsSkillTreeOpen] = React.useState(false);
+
+  const openSkillTree = () => {
+    setIsSkillTreeOpen(true);
+  };
+
+  const open = () => {
+    setIsSkillTreeOpen(false);
+    setIsOpen(true);
+    homeDoorAudio.play();
+  };
+
+  const { gathering, farming } = state.skills;
+
+  const toolLevel = getLevel(gathering);
+  const farmingLevel = getLevel(farming);
+  const totalLevel = toolLevel + farmingLevel;
+
+  const gatheringRequiredXp = getRequiredXpToLevelUp(toolLevel);
+  const farmingRequiredXp = getRequiredXpToLevelUp(farmingLevel);
+
+  const Badges = () => {
+    const BADGES: InventoryItemName[] = [
+      "Green Thumb",
+      "Barn Manager",
+      "Seed Specialist",
+      "Wrangler",
+      "Lumberjack",
+      "Prospector",
+      "Logger",
+      "Gold Rush",
+    ];
+
+    const badges = BADGES.map((badge) => {
+      if (gameState.context.state.inventory[badge]) {
+        return (
+          <img
+            key={badge}
+            src={ITEM_DETAILS[badge].image}
+            alt={badge}
+            className="h-6 mr-2 mb-2 md:mb-0"
+          />
+        );
+      }
+
+      return null;
+    }).filter(Boolean);
+
+    if (badges.length === 0) {
+      return (
+        <span className="text-xs text-shadow">
+          Reach level 5 in a profession to unlock a skill
+        </span>
+      );
+    }
+
+    return <div className="flex flex-wrap">{badges}</div>;
+  };
+
+  const Content = () => {
+    if (isSkillTreeOpen) {
+      return <SkillTree back={open} />;
+    }
+
+    return (
```

Comment:
Consider componentizing this to ensure the visiting and farm House UI remains consistent.

Reply:
I made the `House` reuseable